coffeecard/CoffeeCard.Tests.Unit/Services/v2/ProductServiceTest.cs

```diff
@@ -70,16 +70,14 @@ await productService.UpdateProduct(new UpdateProductRequest()
                 AllowedUserGroups = new List<UserGroup>() { UserGroup.Customer, UserGroup.Board }
             });
 
-            var expected = new List<UserGroup>
-            {
-                UserGroup.Customer, UserGroup.Board
-            };
-
             var result = await productService.GetProductAsync(1);
 
-            Assert.Collection<UserGroup>(expected,
-                e => e.Equals(UserGroup.Customer),
-                e => e.Equals(UserGroup.Board));
+            Assert.Collection<ProductUserGroup>(result.ProductUserGroup,
+                e => Assert.Equal(UserGroup.Customer, e.UserGroup),
+                e => Assert.Equal(UserGroup.Board, e.UserGroup));
+
+            Assert.DoesNotContain(UserGroup.Barista, result.ProductUserGroup.Select(e => e.UserGroup));
+            Assert.DoesNotContain(UserGroup.Manager, result.ProductUserGroup.Select(e => e.UserGroup));
```

Comment:
Doesn't Assert.Collection check that the collection has exactly two elements inside, implying that if it passes, there should be no other elements inside result.ProductUserGroup (making DoesNotContain unnecesary)?

Reply:
It does, but I have left it there for a reason:

Testing is one of the strongest ways of documenting expected behaviour. Changes in this area in the future should be very aware that it is not just expected that the collection contains those two elements in `Assert.Collection` - but they should have removed the two elements from `Assert.DoesNotContain`. In other words, I believe they convey important information - despite not strictly being necessary in terms of the code test.